javaslang/src/main/java/javaslang/collection/BitSet.java

```diff
@@ -0,0 +1,1024 @@
+     */
+    static BitSet<Integer> range(int from, int toExclusive) {
+        return BitSet.ofAll(Iterator.range(from, toExclusive));
+    }
+
+    static BitSet<Character> range(char from, char toExclusive) {
+        return BitSet.withCharacters().ofAll(Iterator.range(from, toExclusive));
+    }
+
+    static BitSet<Long> range(long from, long toExclusive) {
+        return BitSet.withLongs().ofAll(Iterator.range(from, toExclusive));
+    }
+
+    /**
+     * Creates a BitSet of int numbers starting from {@code from}, extending to {@code toExclusive - 1},
+     * with {@code step}.
+     *
+     * @param from        the first number
+     * @param toExclusive the last number + 1
+     * @param step        the step
+     * @return a range of long values as specified or the empty range if<br>
+     * {@code from >= toInclusive} and {@code step > 0} or<br>
+     * {@code from <= toInclusive} and {@code step < 0}
+     * @throws IllegalArgumentException if {@code step} is zero
+     */
+    static BitSet<Integer> rangeBy(int from, int toExclusive, int step) {
+        return BitSet.ofAll(Iterator.rangeBy(from, toExclusive, step));
+    }
+
+    static BitSet<Character> rangeBy(char from, char toExclusive, int step) {
+        return BitSet.withCharacters().ofAll(Iterator.rangeBy(from, toExclusive, step));
+    }
+
+    static BitSet<Long> rangeBy(long from, long toExclusive, long step) {
+        return BitSet.withLongs().ofAll(Iterator.rangeBy(from, toExclusive, step));
+    }
+
+    /**
+     * Creates a BitSet of int numbers starting from {@code from}, extending to {@code toInclusive}.
+     *
+     * @param from        the first number
+     * @param toInclusive the last number
+     * @return a range of int values as specified or the empty range if {@code from > toInclusive}
+     */
+    static BitSet<Integer> rangeClosed(int from, int toInclusive) {
+        return BitSet.ofAll(Iterator.rangeClosed(from, toInclusive));
+    }
+
+    static BitSet<Character> rangeClosed(char from, char toInclusive) {
+        return BitSet.withCharacters().ofAll(Iterator.rangeClosed(from, toInclusive));
+    }
+
+    static BitSet<Long> rangeClosed(long from, long toInclusive) {
+        return BitSet.withLongs().ofAll(Iterator.rangeClosed(from, toInclusive));
+    }
+
+    /**
+     * Creates a BitSet of int numbers starting from {@code from}, extending to {@code toInclusive},
+     * with {@code step}.
+     *
+     * @param from        the first number
+     * @param toInclusive the last number
+     * @param step        the step
+     * @return a range of int values as specified or the empty range if<br>
+     * {@code from > toInclusive} and {@code step > 0} or<br>
+     * {@code from < toInclusive} and {@code step < 0}
+     * @throws IllegalArgumentException if {@code step} is zero
+     */
+    static BitSet<Integer> rangeClosedBy(int from, int toInclusive, int step) {
+        return BitSet.ofAll(Iterator.rangeClosedBy(from, toInclusive, step));
```

Comment:
> With #1295 we will change Iterator, it will not be Iterable any more, so we cannot use BitSet.ofAll(Iterator) in a future version.

I see. But let's do it in future (milestone 3.0.0 I guess) and in own separate commit, m? :smile: